src_cam/mo_chm_diags.F90

```diff
@@ -463,57 +472,100 @@ subroutine chm_diags_inti
-          cloudTracerIndex_direct = getCloudTracerIndexDirect(n)
-          if(cloudTracerIndex_direct > 0)then
-             cloudTracerName(1:len(CloudTracerName))=" "
-             cloudTracerName = getCloudTracerName(n)
-             call addfld( trim(cloudTracerName), (/'lev'/), 'A','kg/kg', &
-                  trim(cloudTracerName)//' in cloud water')
-             call add_default( trim(cloudTracerName), 1, ' ' )
-
-             !Add column burden of cloud tracers
-             call addfld('cb_'//trim(cloudTracerName),horiz_only, 'A', 'kg/m2', &
-                  'cb_'//trim(cloudTracerName)//' column in cloud water')
-             call add_default('cb_'//trim(cloudTracerName),1,' ')
-          endif
-          !..and column burden in clean air
-          call addfld('cb_'//trim(spc_name),horiz_only, 'A', 'kg/m2', &
-               'cb_'//trim(spc_name)//' in column')
-          call add_default('cb_'//trim(spc_name),1,' ' )
-
-          if (history_aerosol)then
-             if (cloudTracerIndex_direct > 0)then
-                !Output budget-terms for cloud borne aerosols
-                call add_default (trim(cloudTracerName)//'GVF', 1, ' ')
-                call add_default (trim(cloudTracerName)//'SFWET', 1, ' ')
-                call add_default (trim(cloudTracerName)//'TBF', 1, ' ')
-                call add_default (trim(cloudTracerName)//'DDF', 1, ' ')
-                call add_default (trim(cloudTracerName)//'SFSBS', 1, ' ')
-                call add_default (trim(cloudTracerName)//'SFSIC', 1, ' ')
-                call add_default (trim(cloudTracerName)//'SFSBC', 1, ' ')
-                call add_default (trim(cloudTracerName)//'SFSIS', 1, ' ')
-             endif
-          endif
-       end if
-       ! OSLO_AERO end
+      ! OSLO_AERO begin
+      ! Add the 3D consentrations 
+      if ( n > 0 ) then 
+         ! if history_aerosol_decomposed we add the aerosol species
+         if ( (any( aer_species == m ) .or. isAerosol(n)) ) then
+            if ( history_aerosol_decomposed ) then
+               call add_default( spc_name, 1, ' ' )
+            endif
+         ! if it is not an aerosol species it is a gas species and we then require the history_gas flagg
+         else 
+            if ( history_gas ) then 
+               call add_default( spc_name, 1, ' ' )
+            endif
+         endif
+      elseif ( trim(spc_name) == 'H2O' ) then
+         if ( history_gas ) then 
+            call add_default( spc_name, 1, ' ' )
+         endif
+      endif 
+
+      if(n > 0) then
+         ! Add cloud tracers to default output
+         cloudTracerIndex_direct = getCloudTracerIndexDirect(n)
+         if ( cloudTracerIndex_direct > 0 ) then
+            ! first the 3d fields, 
+            cloudTracerName(1:len(CloudTracerName))=" "
+            cloudTracerName = getCloudTracerName(n)
+            call addfld( trim(cloudTracerName), (/'lev'/), 'A','kg/kg', &
+               trim(cloudTracerName)//' in cloud water')
```

Comment:
A consequence of putting the `addfld` call _inside_ the `cloudTracerIndex_direct > 0` conditional is that if you add the field manually (say to `fincl2` in user_nl_cam) the field will not be output and in fact, the model will crash during initialization. Is this what is desired (e.g., if the value is only computed  if `cloudTracerIndex_direct > 0`)?

Reply:
Is this really intentional? Are these quantities invalid if `cloudTracerIndex_direct <= 0`?

Reply:
Due to code movement, this comment is no longer relevant, thanks for pointing that out.